### PR TITLE
Build AAR for all build variants by default

### PR DESCRIPTION
**Pull request by blasten**
This will be the new default of `flutter build aar`:

```
$ flutter build aar
Building flutter tool...
Running "flutter pub get" in flutter_module...                      0.4s
Running Gradle task 'assembleAarRelease'...
Running Gradle task 'assembleAarRelease'... Done                   56.4s
✓ Built build/host/outputs/repo.
Running Gradle task 'assembleAarDebug'...
Running Gradle task 'assembleAarDebug'... Done                     29.5s
✓ Built build/host/outputs/repo.
Running Gradle task 'assembleAarProfile'...
Running Gradle task 'assembleAarProfile'... Done                   71.9s
✓ Built build/host/outputs/repo.

Consuming the Module
  1. Open <host>/app/build.gradle
  2. Ensure you have the repositories configured, otherwise add them:

      repositories {
        maven {
            url '/Users/egarciad/p/flutter_apps/add2app/flutter_module/build/host/outputs/repo'
        }
        maven {
            url 'http://download.flutter.io'
        }
      }

  3. Make the host app depend on the Flutter module:

    dependencies {
      releaseImplementation 'com.example.flutter_module:flutter_release:1.0
      debugImplementation 'com.example.flutter_module:flutter_debug:1.0
      profileImplementation 'com.example.flutter_module:flutter_profile:1.0
    }


  4. Add the `profile` build type:

    android {
      buildTypes {
        profile {
          initWith debug
        }
      }
    }

To learn more, visit https://flutter.dev/go/build-aar
```

### CODE REVIEW

**Review by matthew-carroll — APPROVED** [review]
LGTM. Thanks.

File: packages/flutter_tools/lib/src/android/android_builder.dart [hunk 1]
```diff
@@ -26,9 +26,8 @@ abstract class AndroidBuilder {
   /// Builds an AAR artifact.
   Future<void> buildAar({
     @required FlutterProject project,
-    @required AndroidBuildInfo androidBuildInfo,
-    @required String target,
```
Comment (xster) [inline packages/flutter_tools/lib/src/android/android_builder.dart]: This seems unrelated right? We should still keep the ability to specify the main dart file

File: packages/flutter_tools/lib/src/android/android_builder.dart [hunk 2]
```diff
@@ -54,23 +53,30 @@ class _AndroidBuilderImpl extends AndroidBuilder {
...
+          project: project,
+          androidBuildInfo: androidBuildInfo,
+          outputDirectory: outputDirectory,
+        );
+      }
+      printHowToConsumeAar(
+        buildModes: androidBuildInfo
+          .map<String>((AndroidBuildInfo androidBuildInfo) {
+            return androidBuildInfo.buildInfo.modeName;
+          }).toSet(),
```
Comment (xster) [inline packages/flutter_tools/lib/src/android/android_builder.dart]: should we order them? Since we tend to keep debug/profile/release in that order in many other docs etc

File: packages/flutter_tools/lib/src/commands/build_aar.dart [hunk 3]
```diff
@@ -61,21 +61,43 @@ class BuildAarCommand extends BuildSubCommand {
 
   @override
   final String description = 'Build a repository containing an AAR and a POM file.\n\n'
-      'The POM file is used to include the dependencies that the AAR was compiled against.\n\n'
+      'By default, AARs are built for `release`, `debug` and `profile`.\n'
```
Comment (xster) [inline packages/flutter_tools/lib/src/commands/build_aar.dart]: Is it possible to make it not do the default? Couldn't tell from reading the implementation. The iOS one has flags like https://github.com/flutter/flutter/blob/01f4f1ac5576a09dca32df6a2c9eaca1af1a7f22/packages/flutter_tools/lib/src/commands/build_ios_framework.dart#L41

**Review by xster — COMMENTED** [review]
LGTM. 

@jmagman, it looks like your way has prevailed :)

File: packages/flutter_tools/lib/src/android/android_builder.dart (hunk 1 above)
Comment (blasten) [reply packages/flutter_tools/lib/src/android/android_builder.dart]: oops. reverted
